### PR TITLE
Boost => 1.82, Gimp => 2.99.16 & deps

### PR DESCRIPTION
**Pull request by satmandu**
Updates:
- [x] gnulib_git
- [x] Icu4c => 73.2
- [x] Boost => 1.82 and the packages that depend upon boost:
  - [x] Source_highlight
  - [x] poppler => 23.07
  - [x] eigen
  - [x] exempi
  - [x] gdb
  - [x] grive
    - [x] yajl
    - [x] libgpgerror
    - [x] libgcrypt 
    - [x] gnupg
  - [x] ledger
  - [x] libcdr
  - [x] librevenge
  - [x] libtorrent
  - [x] libvisio
  - [x] inkscape (Runs properly!)
  - [x] swig
  - [x] taglib
  - [x] tcpflow
- [x] Babl
- [x] mold (update to fix [LTO compile issue](https://github.com/rui314/mold/commit/4b42f38257068f2a3f0dbb102904519d85c9dcb2).)
- [x] Gegl
- [x] Gimp => 2.99.16 (Runs properly!)

Builds properly:
- [x] `x86_64`
- [x] `i686`
- [x] `armv7l` 

### Run the following to get this pull request's changes locally for testing.
```
CREW_TESTING_REPO=https://github.com/satmandu/chromebrew.git CREW_TESTING_BRANCH=gimp29916 CREW_TESTING=1 crew update && crew upgrade
```

### CODE REVIEW

**Comment by satmandu** [comment]
gdb for some reason appears to have been built against an older binutils... :/

**Comment by satmandu** [comment]
ok, building binutils rebuild which works around this issue with gdb...

**Comment by satmandu** [comment]
gdb works now.

File: manifest/armv7l/g/gimp.filelist [hunk 1]
```diff
@@ -1,16 +1,11 @@
-/usr/local/bin/gimp
 /usr/local/bin/gimp-2.99
```
Comment (uberhacker) [inline manifest/armv7l/g/gimp.filelist:1]: Does this mean in order to start we have to type gimp-2.99?  Probably should restore if these are symlinks.

Comment (satmandu) [reply manifest/armv7l/g/gimp.filelist:1]: If you hit gimp <tab> it autocompletes to gimp-2.99.

As such, should we add symlinks to gimp or not?

File: packages/gdb.rb [hunk 2]
```diff
@@ -39,44 +40,7 @@ class Gdb < Package
...
-
-      -  return rl_completer_word_break_characters;
-      +  return (char *) rl_completer_word_break_characters;
-       }
-
-       char *
-    READLINE8_PATCH_EOF
-    File.write('upstream-fix-build-readline8.patch', @readline8patch)
-    system 'patch -Np1 -i upstream-fix-build-readline8.patch'
-  end
```
Comment (uberhacker) [inline packages/gdb.rb:79]: Another patch bites the dust.

File: packages/poppler.rb [hunk 3]
```diff
@@ -3,25 +3,25 @@
 class Poppler < Package
```
Comment (uberhacker) [inline packages/poppler.rb:3]: `class Poppler < CMake`?

File: packages/taglib.rb [hunk 4]
```diff
@@ -3,35 +3,35 @@
 class Taglib < Package
```
Comment (uberhacker) [inline packages/taglib.rb:3]: `class Taglib < CMake` with sed in patch section?

File: packages/tcpflow.rb [hunk 5]
```diff
@@ -3,29 +3,29 @@
 class Tcpflow < Package
```
Comment (uberhacker) [inline packages/tcpflow.rb:3]: `class Tcpflow < Autotools`?

File: packages/yajl.rb [hunk 6]
```diff
@@ -3,23 +3,23 @@
 class Yajl < Package
```
Comment (uberhacker) [inline packages/yajl.rb:3]: `class Yajl < CMake`?

**Review by uberhacker — CHANGES_REQUESTED** [review]
It is possible to build some of these with buildsystems?

File: packages/poppler.rb (hunk 3 above)
Comment (satmandu) [reply packages/poppler.rb:3]: No because of the `cmake_option` `-DENABLE_UNSTABLE_API_ABI_HEADERS=ON \`

File: packages/taglib.rb (hunk 4 above)
Comment (satmandu) [reply packages/taglib.rb:3]: wasn't working. hence the non-standard build options.

File: packages/tcpflow.rb (hunk 5 above)
Comment (satmandu) [reply packages/tcpflow.rb:3]: uses non-standard option ` system 'bash bootstrap.sh'`

File: packages/yajl.rb (hunk 6 above)
Comment (satmandu) [reply packages/yajl.rb:3]: uses `CREW_CMAKE_LIBSUFFIX_OPTIONS`...

**Review by uberhacker — APPROVED** [review]

**Pull request merged**